plusdeck/cli.py

```diff
@@ -0,0 +1,371 @@
+                try:
+                    client: Client = await create_connection(port)
+                except SerialException as exc:
+                    click.echo(exc)
+                    sys.exit(1)
+
+                # Giddyup!
+                try:
+                    async with asyncio.timeout(timeout):
+                        await fn(client, *args, **kwargs)
+                except TimeoutError:
+                    echo(f"Command timed out after {timeout} seconds.")
+                    sys.exit(1)
+
+                # Close the client if we're done
+                if not run_forever:
+                    client.close()
+
+                # Await the client closing and surface any exceptions
+                await client.closed
+
+            try:
+                asyncio.run(main())
+            except KeyboardInterrupt:
+                pass
+
+        return wrapped
+
+    return decorator
+
+
+@click.group()
+@click.option(
+    "--global/--no-global",
+    "global_",
+    default=os.geteuid() == 0,
+    help=f"Load the global config file at {GLOBAL_FILE} "
+    "(default true when called with sudo)",
+)
+@click.option("--config-file", "-C", type=click.Path(), help="A path to a config file")
+@click.option(
+    "--log-level",
+    envvar="CRYSTALFONTZ_LOG_LEVEL",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+    default="INFO",
+    help="Set the log level",
+)
+@click.option(
+    "--port",
+    envvar="CRYSTALFONTZ_PORT",
+    help="The serial port the device is connected to",
+)
+@click.option(
+    "--output",
+    type=click.Choice(["text", "json"]),
+    default="text",
+    help="Output either human-friendly text or JSON",
+)
+@click.option(
+    "--timeout",
+    type=float,
+    envvar="CRYSTALFONTZ_TIMEOUT",
+    help="How long to wait for a response from the device before timing out",
+)
+@click.pass_context
+def main(
+    ctx: click.Context,
+    global_: bool,
+    config_file: Optional[str],
+    log_level: LogLevel,
+    port: Optional[str],
+    output: Optional[OutputMode],
+    timeout: Optional[float],
+) -> None:
+    """
+    Control your Crystalfontz device.
+    """
+
+    file = None
+    if config_file:
+        if global_:
+            warnings.warn(
+                "--config-file is specified, so --global flag will be ignored."
+            )
+        file = config_file
+    elif global_:
+        file = GLOBAL_FILE
+    config: Config = Config.from_file(file=file)
+    ctx.obj = Obj(
+        config=config,
+        global_=global_,
+        port=port or config.port,
+        output=output or "text",
+        timeout=timeout or config.timeout,
+    )
+
+    logging.basicConfig(level=getattr(logging, log_level))
+
+
+@main.group
+def play() -> None:
+    """
+    Play a tape
+    """
+
+
+@play.command(name="a")
+@pass_client()
+async def play_a(client: Client) -> None:
+    """
+    Play side A of the tape
+    """
+
+    client.play_a()
```

Comment:
Is there a chance the client gets closed before the command is sent? Is there a way to ensure that the send side is drained before disconnecting? But test this - it may not be a real concern.

Reply:
This appears to truly be synchronous - or at least, the commands work flawlessly.